iterative/gcp/provider.go

```diff
@@ -0,0 +1,487 @@
+					"22",
+				},
+			},
+		},
+		SourceRanges: []string{
+			"0.0.0.0/0",
+		},
+	}
+
+	firewallIngressInsertOperation, err := service.Firewalls.Insert(project, firewallIngressDefinition).Do()
+	if err != nil {
+		return err
+	}
+
+	firewallIngressGetOperationCall := service.GlobalOperations.Get(project, firewallIngressInsertOperation.Name)
+	_, err = waitForOperation(ctx, d.Timeout(schema.TimeoutCreate), firewallIngressGetOperationCall.Do)
+	if err != nil {
+		return err
+	}
+
+	instanceDefinition := &gcp_compute.Instance{
+		Name:        instanceName,
+		MachineType: instanceMachineType.SelfLink,
+		Disks: []*gcp_compute.AttachedDisk{
+			{
+				Boot:       true,
+				AutoDelete: true,
+				Type:       "PERSISTENT",
+				Mode:       "READ_WRITE",
+				InitializeParams: &gcp_compute.AttachedDiskInitializeParams{
+					DiskName:    instanceName,
+					SourceImage: instanceImage.SelfLink,
+					DiskSizeGb:  instanceHddSize,
+					DiskType:    instanceDiskType.SelfLink,
+				},
+			},
+		},
+		NetworkInterfaces: []*gcp_compute.NetworkInterface{
+			{
+				Network: network.SelfLink,
+				AccessConfigs: []*gcp_compute.AccessConfig{
+					{
+						NetworkTier: "STANDARD",
+					},
+				},
+			},
+		},
+		Tags: &gcp_compute.Tags{
+			Items: []string{instanceName},
+		},
+		Scheduling: &gcp_compute.Scheduling{
+			OnHostMaintenance: instanceHostMaintenanceBehavior,
+			Preemptible:       instanceIsPreemptible,
+		},
+		Metadata: &gcp_compute.Metadata{
+			Items: []*gcp_compute.MetadataItems{
+				{
+					Key:   "ssh-keys",
+					Value: &instancePublicSshKey,
+				},
+				{
+					Key:   "startup-script",
+					Value: &instanceStartupScript,
+				},
+			},
+		},
+		GuestAccelerators: instanceAccelerators,
+	}
+
+	instanceInsertOperation, err := service.Instances.Insert(project, instanceZone, instanceDefinition).Do()
+	if err != nil {
+		return err
+	}
+
+	instanceGetOperationCall := service.ZoneOperations.Get(project, instanceZone, instanceInsertOperation.Name)
+	_, err = waitForOperation(ctx, d.Timeout(schema.TimeoutCreate), instanceGetOperationCall.Do)
+	if err != nil {
+		return err
+	}
+
+	instance, err := service.Instances.Get(project, instanceZone, instanceName).Do()
+	if err != nil {
+		return err
+	}
+
+	d.Set("instance_ip", instance.NetworkInterfaces[0].AccessConfigs[0].NatIP)
+	d.Set("instance_launch_time", time.Now().String())
+
+	return nil
+}
+
+func ResourceMachineDelete(ctx context.Context, d *schema.ResourceData, m interface{}) error {
+	project, service, err := getProjectService()
+	if err != nil {
+		return err
+	}
+
+	instanceZone := getRegion(d.Get("region").(string))
+	instanceName := d.Id()
+
+	instanceDeleteOperationCall := service.Instances.Delete(project, instanceZone, instanceName)
+	_, err = waitForOperation(ctx, d.Timeout(schema.TimeoutDelete), instanceDeleteOperationCall.Do)
+	if err != nil {
+		return err
+	}
+
+	firewallIngressDeleteOperationCall := service.Firewalls.Delete(project, instanceName+"-ingress")
```

Comment:
I do not think this is going to happen if the runner disposes the machine, we should use instance-resources to just call one delete

Reply:
Now this code fails every single time

```sh
Error: Failed disposing the machine: googleapi: Error 404: The resource 'projects/cml-gcp-test/zones/us-west1-b/instances/iterative-18m2033bvd8tx' was not found, notFound
```